packages/core/src/browser/tree/tree-view-welcome-widget.tsx

```diff
@@ -0,0 +1,245 @@
+/********************************************************************************
+ * Copyright (c) 2020 SAP SE or an SAP affiliate company and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import React = require('react');
+import { inject, injectable } from 'inversify';
+import { CommandRegistry } from '../../common';
+import { ContextKeyService } from '../context-key-service';
+import { TreeModel } from './tree-model';
+import { TreeWidget } from './tree-widget';
+
+interface ViewWelcome {
+    view: string;
+    content: string;
+    when?: string;
+    order: number;
+}
+
+interface IItem {
+    readonly welcomeInfo: ViewWelcome;
+    visible: boolean;
+}
+
+interface ILink {
+    readonly label: string;
+    readonly href: string;
+    readonly title?: string;
+}
```

Comment:
This PR requires a CQ for sure:

https://github.com/microsoft/vscode/blob/573e5145ae3b50523925a6f6315d373e649d1b06/src/vs/base/common/linkedText.ts#L8-L12


@marcdumais-work, @amiramw can one of you please help @EstherPerelman with the CQ submission? The changeset works pretty great. Theia needs this addition but we have to clearly document, the idea was taken from VS Code. Thank you!

Reply:
@kittaakos i have never dealt with CQ. is the process documented somewhere?

Reply:
Yes, it is documented on the Wiki. [Here](https://github.com/eclipse-theia/theia/wiki/Registering-CQs#codecontent-is-copied-from-the-main-vs-code-repo) is the relevant section when the code was copied from VS Code.

Reply:
@amiramw have you opened the CQ? With the increased contributions, it's good that one SAP committer joins the ranks of our "IP champions". Let me know if I have questions. 

If by chance you can find the code you want, already covered by a previous, approved CQ, then we would not need a new one. For example, we often request to copy from `vscode.d.ts`.

Reply:
@marcdumais as i understand since this copied code is from newer version then i need to open a CQ based on https://github.com/eclipse-theia/theia/wiki/Registering-CQs#codecontent-is-copied-from-other-projectsrepos, right?

Where can I see all previous CQs?

Reply:
@kittaakos maybe you can answer?

Reply:
> Where can I see all previous CQs?

No idea. I bet filtering by label does not show all: https://github.com/eclipse-theia/theia/pulls?q=+is%3Apr+label%3A%22CQ+Required%22+

Reply:
> @marcdumais as i understand since this copied code is from newer version then i need to open a CQ based on https://github.com/eclipse-theia/theia/wiki/Registering-CQs#codecontent-is-copied-from-other-projectsrepos, right?
> 

Yes, I think so. The exception being if the code has been there for a while, in a form you can use, then you are allowed to pick the code from whatever version you want.

> Where can I see all previous CQs?

I am no expert, but I start [here](https://dev.eclipse.org/ipzilla/query.cgi) and usually only do a simple keywork search, e.g. for [vscode](https://dev.eclipse.org/ipzilla/buglist.cgi?query_format=advanced&short_desc_type=allwordssubstr&short_desc=vscode&long_desc_type=substring&long_desc=&bug_file_loc_type=allwordssubstr&bug_file_loc=&keywords_type=allwords&keywords=&emailassigned_to1=1&emailtype1=substring&email1=&emailassigned_to2=1&emailreporter2=1&emailcc2=1&emailtype2=substring&email2=&bugidtype=include&bug_id=&chfieldfrom=&chfieldto=Now&chfieldvalue=&cmdtype=doit&order=Reuse+same+sort+as+last+time&field0-0-0=noop&type0-0-0=noop&value0-0-0=) or whatever I am looking for.

Reply:
Has anyone already registered the CQ? I do not want to hold back this great change. So if this PR is blocked only by me, let me know.

Reply:
@kittaakos @marcdumais-work I registered CQ https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22793 now. It's my first time so I hope someone can take a look at it.

Reply:
Great! Thank you so much.

I looked into the attachment, and it looks good to me. I would have created the same archive.🤞

Reply:
@kittaakos the CQ is approved now